### PR TITLE
MV3: fix reporing startup race

### PR DESCRIPTION
**Pull request by chrmod**
Alternative for https://github.com/ghostery/ghostery-extension/pull/1137 and #1115

### CODE REVIEW

File: extension-manifest-v3/src/background/utils/startup-guard.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+export default (starter, stopper) => {
```
Comment (smalluban) [inline extension-manifest-v3/src/background/utils/startup-guard.js:1]: If you want to have the option to abort a "pending" promise, we can use the `AbortController` API, as it is already widely supported: https://caniuse.com/abortcontroller (including Safari, which is our main target currently).

Comment (smalluban) [inline extension-manifest-v3/src/background/utils/startup-guard.js:1]: Please use the `function` statement in modules where possible (if you don't set `const something = () => {}`. Also, the copyright missing.

**Review by smalluban — CHANGES_REQUESTED** [review]
Generally, this solution only adds a feature to switch the result of the setup process, so it is mutable (not like in #1152, where once finished, the promise is cleared).

However, it does not help with the reporting much, as we could simply check `reporting.isActive` as before. When `terms` go to `false` (which is only possible by dev tools, which clears out the options), observe method can just call `reporting.unload()` - which is sync operation by the way.

I can't see here a case, which cannot be covered with #1152

**Comment by chrmod** [comment]
fixed by https://github.com/ghostery/ghostery-extension/pull/1152

**Pull request closed without merging**